### PR TITLE
build: python 3 compatibility

### PR DESCRIPTION
**Pull request by laanwj**
Ubuntu 16.04 "xenial xerus" does not come with Python 2.x by default. It is possible to install a python-2.7 package, but this has its own problem: no `python` or `python2` symlink (see #7717).

This fixes the following scripts to work with python 3:
- `make check` (bctest,py, bitcoin-util-test.py)
- `make translate` (extract_strings_qt.py)
- `make symbols-check` (symbol-check.py)
- `make security-check` (security-check.py)
- `make deploy` for OS X (custom_dsstore.py, macdeployqtplus)

Explicitly call the python commands using $(PYTHON) (detected by autoconf) instead of relying on the interpreter line at the top of the scripts.

Python 2.x compatibility should be unaffected. For the build system I think it's good to have both Python2 and Python3 compatibility. This is not necessary for the other python scripts, such as the RPC tests. For practical reasons it's ok to have an explicit Python 2.7 dependency for those - though we'll have to document that.

### CODE REVIEW

**Comment by laanwj** [comment]
Ping @theuni

**Comment by MarcoFalke** [comment]
Looks good. Concept ACK  40a9d86

**Comment by jonasschnelli** [comment]
Concept ACK.

Why not use `!/usr/bin/env python` as interpreter?
IMO we should at least remove /usr/bin/python2 and use /usr/bin/python (https://github.com/bitcoin/bitcoin/pull/7723/files#diff-d403511be890783565b1e2d164beec8aR1).

**Comment by laanwj** [comment]
> Why not use !/usr/bin/env python as interpreter?

Interpeter shouldn't matter anymore because all usages invoke the explicitly located python executable.

**Comment by theuni** [comment]
@laanwj Aha, clever fix :)

 I'm not well-versed in python2/python3 to comment on the compat there, but concept ack for sure.

Looks like some of the mac deploy scripts are missing, though:
```
contrib/macdeploy/custom_dsstore.py
contrib/macdeploy/macdeployqtplus
```
Those both get hit via ```make deploy```

**Comment by laanwj** [comment]
```
contrib/macdeploy/custom_dsstore.py
contrib/macdeploy/macdeployqtplus
```
Right, good catch, will have a look at those.

**Comment by laanwj** [comment]
@cfields see b40e365c2a24019e2b3e2651510874d0cecf2ae3
Works in python2 at least, haven't tested w/ python3 yet, I don't have a macosx dev environment set up.

**Comment by laanwj** [comment]
> Works in python2 at least, haven't tested w/ python3 yet, I don't have a macosx dev environment set up.

Ok, tested by temporarily changing the detection to only 'see' python3. It passed.

File: contrib/macdeploy/macdeployqtplus [hunk 1]
```diff
@@ -575,7 +575,7 @@ if len(config.fancy) == 1:
     if fancy.has_key("background_picture"):
```
Comment (MarcoFalke) [inline contrib/macdeploy/macdeployqtplus]: Nit: there is no `has_key` in py3

**Comment by MarcoFalke** [comment]
utACK 	b40e365

File: contrib/macdeploy/macdeployqtplus (hunk 1 above)
Comment (laanwj) [inline contrib/macdeploy/macdeployqtplus]: Good catch

**Comment by laanwj** [comment]
Squashed b875273, b40e365 and cb01698 into 18f05c7

**Pull request merged**